rotkehlchen/chain/evm/tokens.py

```diff
@@ -220,13 +249,18 @@ def detect_tokens(
         - BadFunctionCallOutput if a local node is used and the contract for the
           token has no code. That means the chain is not synced
         """
+        # Also detect tokens for proxies
+        addresses, proxies_mapping = self._proxies_for_addresses(addresses)
+
         with self.db.conn.read_ctx() as cursor:
             if only_cache is False:
                 self._detect_tokens(cursor, addresses=addresses)
 
             addresses_info: DetectedTokensType = {}
             for address in addresses:
-                addresses_info[address] = self.db.get_tokens_for_address(
+                # If `address` is a proxy, save token balances to its owner address
+                proxied_address = proxies_mapping.get(address, address)
+                addresses_info[proxied_address] = self.db.get_tokens_for_address(
```

Comment:
isn't this skipping information? since you need to query the tokens for `address` and `proxied_address`. Not only the proxied address

Reply:
Yes, exactly, thanks! It does not happen with current code since in the current code token balances for proxies and not-proxies are queried separately, so they don't overlap. But I'll fix it and adjust the test